modeler/cayenne-modeler/src/main/java/org/apache/cayenne/modeler/editor/dbimport/DbImportSorter.java

```diff
@@ -0,0 +1,110 @@
+
+    public static void sortNode(DbImportTreeNode node) {
+        sortNodeItems(node);
+        syncUserObjectItems(node);
+    }
+
+    public static void sortNodeWithAllChildren(DbImportTreeNode node) {
+        DbImportSorter.sortNode(node);
+        for (int i = 0; i < node.getChildCount(); i++) {
+            sortNodeWithAllChildren((DbImportTreeNode) node.getChildAt(i));
+        }
+    }
+
+    private static void sortNodeItems(DbImportTreeNode node) {
+        List<DbImportTreeNode> childNodes = node.getChildNodes();
+        node.removeAllChildren();
+        childNodes.sort(NODE_COMPARATOR);
+        childNodes.forEach(node::add);
+    }
+
+    private static void syncUserObjectItems(DbImportTreeNode parentNode) {
+
+        Object userObject = parentNode.getUserObject();
+
+        if (parentNode.isReverseEngineering()) {
+            syncCatalogs(parentNode, (ReverseEngineering) userObject);
+            syncSchemas(parentNode, (ReverseEngineering) userObject);
+            syncPatternParams(parentNode, (ReverseEngineering) userObject);
+        }
+        if (parentNode.isCatalog()) {
+            syncSchemas(parentNode, (Catalog) userObject);
+            syncPatternParams(parentNode, (Catalog) userObject);
+        }
+        if (parentNode.isSchema()) {
+            syncPatternParams(parentNode, (Schema) userObject);
+        }
+        if (parentNode.isIncludeTable()) {
+            syncTablesColumns(parentNode, (IncludeTable) userObject);
+        }
+    }
+
+    private static void syncCatalogs(DbImportTreeNode node, ReverseEngineering reverseEngineering) {
+        syncItemsInContainer(reverseEngineering.getCatalogs(),node,NodeType.CATALOG);
+    }
+
+    private static void syncSchemas(DbImportTreeNode node, SchemaContainer schemaContainer) {
+        syncItemsInContainer(schemaContainer.getSchemas(),node,NodeType.SCHEMA);
+    }
+
+    private static void syncPatternParams(DbImportTreeNode node, FilterContainer container) {
+        syncItemsInContainer(container.getIncludeTables(),node,NodeType.INCLUDE_TABLE);
+        syncItemsInContainer(container.getExcludeTables(),node,NodeType.EXCLUDE_TABLE);
+        syncItemsInContainer(container.getIncludeColumns(),node,NodeType.INCLUDE_COLUMN);
+        syncItemsInContainer(container.getExcludeColumns(),node,NodeType.EXCLUDE_COLUMN);
+        syncItemsInContainer(container.getIncludeProcedures(),node,NodeType.INCLUDE_PROCEDURE);
+        syncItemsInContainer(container.getExcludeProcedures(),node,NodeType.EXCLUDE_PROCEDURE);
+    }
+
+    private static void syncTablesColumns(DbImportTreeNode node, IncludeTable table) {
+        syncItemsInContainer(table.getIncludeColumns(),node,NodeType.INCLUDE_COLUMN);
+        syncItemsInContainer(table.getExcludeColumns(),node,NodeType.EXCLUDE_COLUMN);
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private static void syncItemsInContainer(Collection collection, DbImportTreeNode node, NodeType type){
```

Comment:
It could be better to add a generic parameter, like `<T> sync(Collection<T>...)`

Reply:
Accepted and done